### PR TITLE
fix flaky log rotation test

### PR DESCRIPTION
**Pull request by ashmeenkaur**
### Description
There is an edge case that rarely happens in log rotation integration tests where we try to stat a log file when the logs are getting rotated (the old log file getting renamed and new log file being created). This was causing failure in the integration tests. In order to fix the issue, we will retry the stat call once again before failing the test. If two consecutive  stat calls fail, we will fail the test.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Ran as part of presubmit.

### CODE REVIEW

File: tools/integration_tests/log_rotation/logrotate_logfile_test.go [hunk 1]
```diff
@@ -62,7 +63,12 @@ func runOperationsOnFileTillLogRotation(t *testing.T, wg *sync.WaitGroup, fileNa
 		// Break the loop when log file is rotated.
 		fi, err := operations.StatFile(logFilePath)
 		if err != nil {
-			t.Errorf("stat operation on file %s: %v", logFilePath, err)
+			t.Logf("stat operation on file %s failed: %v", logFilePath, err)
+			if !retryStatLogFile {
```
Comment (raj-prince) [inline tools/integration_tests/log_rotation/logrotate_logfile_test.go:67]: The issue might come again in case when rotation will happen twice for a file. 
I think, we should add a check to catch when error will happen back to back (in consecutive loops).

Comment (ashmeenkaur) [reply tools/integration_tests/log_rotation/logrotate_logfile_test.go:67]: Discussed offline that this is only for one log rotation operation, so this check won't be required.
